File: 1801-1900/1894-Chalk-Replacer/1894.py
class Solution:
  def chalkReplacer(self, chalk: List[int], k: int) -> int:
    total = sum(chalk)
    k %= total
    res = -1
    for i, cnt in enumerate(chalk):
      if cnt > k:
        res = i
        break
      k -= cnt
    return res

File: 1801-1900/1894-Chalk-Replacer/1894.java
class Solution {
  public int chalkReplacer(int[] chalk, int k) {
    int n = chalk.length;
    long total = 0;
    for (int num : chalk) {
      total += num;
    }
    k %= total;
    int res = -1;
    for (int i = 0; i < n; ++i) {
      if (chalk[i] > k) {
        res = i;
        break;
      }
      k -= chalk[i];
    }
    return res;
  }
}